### DISCUSSION TITLE
Can't read cbor files

### DISCUSSION THREAD

**Discussion by HamletDuFromage** [nlohmann/json#2669] [Q&A]
Hello,
I'm trying to read CBOR files but I'm having trouble with it. I wrote a little snippet that creates a cbor and then attempts to get the json from it

```C++
json test = R"({"compact": true, "schema": 0})"_json;
std::ofstream output_file("/test.cbor", std::ios::out | std::ios::binary);
std::vector<uint8_t> output_vector;
json::to_cbor(test, output_vector);
output_file.write((char*)output_vector.data(), output_vector.size());

std::ifstream input_file("/test.cbor", std::ios::binary);
std::vector<uint8_t> input_vector((std::istreambuf_iterator<char>(input_file)), std::istreambuf_iterator<char>());

try { 
    json j = json::from_cbor(input_vector);
}
catch (json::parse_error& e)
{
// output exception information
    std::cout << "message: " << e.what() << '\n'
                << "exception id: " << e.id << '\n'
                << "byte position of error: " << e.byte << std::endl;
}
```

I'm getting this exception:

```
message: [json.exception.parse_error.110] parse error at byte 1: syntax error while parsing CBOR value: unexpected end of input
exception id: 110
byte position of error: 1
```

You can find attached the cbor file, which I can read decode fine with cbor2 on Python
[test.cbor.zip](https://github.com/nlohmann/json/files/6111413/test.cbor.zip)

Does anyone know what the issue may be? I'm guessing this line `output_file.write((char*)output_vector.data(), output_vector.size());` is the culprit, and especially `(char*)` but it doesn't compile without it.

Thanks!

**Comment by scifcha** [comment]
i have the same problem, help would be welcome

**Comment by nlohmann** [comment]
Such an error message usually indicates that the file was not opened successfully and reading the first byte already failed. The slash in the beginning of the filename looks odd. Please make sure the file exists at that path and is actually readable to your program.

**Reply by scifcha** [reply]
> 
> 
> Such an error message usually indicates that the file was not opened successfully and reading the first byte already failed. The slash in the beginning of the filename looks odd. Please make sure the file exists at that path and is actually readable to your program.

yes, everything is fine. only thing i can add is: issue goes away if your json is too small(something like <30 bytes)

**Reply by nlohmann** [reply]
Can you compile and run the unit tests?

**Reply by scifcha** [reply]
I can't do it right now, but I can add that switching to msgpack completely fixes my problem

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?
